### PR TITLE
[1840] Initial auction round and first stock round

### PR DESCRIPTION
**Pull request by Zwergenpunk**
Also adds 
- initial Token placement of city corporations
- logos
- some map fixes

### CODE REVIEW

File: lib/engine/game/g_1840/game.rb [hunk 1]
```diff
@@ -24,26 +24,114 @@ class Game < Game::Base
...
+
+            initial_coordinates_info.each do |info|
+              token = corporation.find_token_by_type
+              city_index = info[:city_index] || 0
+              hex_by_id(info[:coordinate]).tile.cities[city_index].place_token(corporation, token,
+                                                                               check_tokenable: false)
+            end
+          end
+
+          @corporations.clear
```
Comment (roseundy) [inline lib/engine/game/g_1840/game.rb:111]: How does this work when a tram corp appears in an Action, since actions use corporations_by_id ?

Comment (Zwergenpunk) [reply lib/engine/game/g_1840/game.rb:111]: I don't really know. I'll fix it then if needed I guess 🤷 
But thanks for the info

File: lib/engine/game/g_1840/game.rb [hunk 2]
```diff
@@ -24,26 +24,114 @@ class Game < Game::Base
...
+            { coordinate: 'I1' },
+            { coordinate: 'I9' },
+            { coordinate: 'I15' },
+            { coordinate: 'F24' },
+          ],
+          'V' => [
+            { city_index: 1, coordinate: 'A17' },
+            { coordinate: 'A13' },
+            { coordinate: 'B10' },
+            {  coordinate: 'C7' },
```
Comment (tobymao) [inline lib/engine/game/g_1840/game.rb:71]: format this?

File: lib/engine/game/g_1840/step/buy_sell_par_shares.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        class BuySellParShares < Engine::Step::BuySellParShares
+          def actions(entity)
+            return [] unless entity == current_entity
+
+            actions = []
+            actions << 'buy_shares' if can_buy_any?(entity)
+            actions << 'par' if can_ipo_any?(entity)
+            actions << 'sell_shares' if can_sell_any?(entity)
+
+            actions << 'pass' unless actions.empty? || !@game.all_major_corporations_ipoed?
```
Comment (tobymao) [inline lib/engine/game/g_1840/step/buy_sell_par_shares.rb:18]: don't use unless with multiple conditions

File: lib/engine/game/g_1840/step/selection_auction.rb [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+          end
+
+          def starting_bid(company)
+            company.value
+          end
+
+          def min_bid(company)
+            return unless company
+
+            return starting_bid(company) unless @bids[company].any?
```
Comment (tobymao) [inline lib/engine/game/g_1840/step/selection_auction.rb:89]: if empty

File: lib/engine/game/g_1840/step/buy_sell_par_shares.rb (hunk 3 above)
Comment (Zwergenpunk) [reply lib/engine/game/g_1840/step/buy_sell_par_shares.rb:18]: hehe, yeah that is borderline readable.

**Pull request merged**